[Feature] 시간표 뷰에 필요한 장소 컴포넌트 구현

HJunng opened the pull request:
## 📝 변경 사항
시간표 뷰에 장소와 시간 정보를 표시하는 PlaceTimeCard 컴포넌트를 구현했습니다.

## 🎯 작업 내용
- Place 모델의 시간 관련 필드명 변경
- Place 더미 데이터 구체화
  - `createPlace` 헬퍼 함수를 추가하여 더미 데이터 생성
  - 각 장소에 주소, 시작/종료 시간, 카테고리 등 구체적인 정보 추가
  - `createPlaceImages` 함수 분리
- PlaceTimeCard 컴포저블 구현
  - 소요 시간에 따라 90분 미만일 경우 `CompactPlaceTimeCard`, 90분 이상일 경우 `ExpandedPlaceTimeCard` 표시
  - 장소 이름, 시간, 카테고리 아이콘을 포함한 카드 UI 구현

## 🔗 관련 이슈
Closes #43

## 📸 스크린샷 
<img width="282" height="134" alt="image" src="https://github.com/user-attachments/assets/9a464cfc-7484-48bb-951b-fb4bc7b29a21" />



## 💬 리뷰어에게

1. PlaceTimeCard를 보여줄 때 90분 미만/이상 경우를 나눠서 사진을 크게 보여주고, 작게 보여주는 걸 이전에 얘기 나왔던 것 같아서 구현했습니다. 시간표 뷰까지 다 만들고 그거에 맞게 크게/작게 보여주는 기준을 적용해볼 수 있을 것 같아서, 현재는 90분을 기준으로 잡아두었습니다.

2. 1번 사진처럼 구현해서 태그별로 색상들을 잘 활용해보고 싶었는데, 예쁜 거랑 상관없이 깔끔하고 직관적으로 보이게 하려면 2번이 좋을 것 같아서 2번으로 수정은 해두었습니다. 다른 분들은 두가지 중에 어떤 걸로 적용하는게 더 마음에 드시나요??  -> 2번 확정.

HJunng committed:
feat: StaticChip 컴포저블 구현

- 고정된 형태의 태그를 표시하기 위한 `StaticChip` 컴포저블 구현.
- `StaticChipColors` 데이터 클래스를 정의하여 칩의 배경, 텍스트, 테두리 색상을 변경할 수 있도록 구현.

HJunng committed:
feat: CategoryChip 컴포저블 구현

- `StaticChip`을 재사용하여 `CategoryChip` 컴포저블 구현
- 카테고리의 color에 따라 글자 색상을 설정하고, alpha = 0.1f 로 배경색을 지정하도록 구현

HJunng committed:
feat: API desugaring 라이브러리 추가

- java.time 라이브러리를 사용하기 위해 desugaring을 활성화하고 관련 의존성을 추가

HJunng committed:
feat: Place 모델에 상세 정보 필드 추가

- Place 데이터 클래스에 주소, 시작/종료 시간, 카테고리 목록을 추가
- 소요 시간(분)을 계산하는 `durationMinutes` 속성 추가
- 여러 날에 걸친 일정인지 확인하는 `isMultiDay` 속성 추가

HJunng committed:
feat: 시간 및 날짜 형식 문자열 리소스 추가

- 오전/오후, 날짜 및 시간 형식, 시간 형식, 범위 형식을 위한 문자열 리소스 추가

HJunng committed:
feat: desugaring 라이브러리 추가

- `java.time` 라이브러리를 사용하기 위해 desugaring을 활성화

HJunng committed:
feat: MemoripImage 컴포저블 구현

- Coil3의 AsyncImage를 기반으로 한 MemoripImage 컴포저블 추가
- 이미지 로딩 중, 에러 발생 시, 또는 fallback 상태일 때 `offWhite` 색상을 플레이스홀더로 사용하도록 설정

HJunng committed:
feat: 시간 관련 확장 함수 추가

- `LocalDateTime`을 "오후 03:00" 형식의 문자열로 변환하는 `toTimeString()` 확장 함수 추가
- 두 `LocalDateTime` 객체를 "오후 03:00 - 오후 07:00" 형식으로 변환하는 `formatTimeRange()` 함수 추가

HJunng committed:
feat: 다크모드 색상 및 아이콘 크기 추가

- 다크모드에서 사용될 색상 `Black`, `White`를 서로 교체
- 아이콘 크기 `IconSizeSmall` 추가
- 기존 `IconSizeSmall`은 `IconSizeXSmall`로 이름 변경

HJunng committed:
fix: Place 모델의 시간 관련 필드명 변경

HJunng committed:
feat: Place 더미 데이터 구체화

- `createPlace` 헬퍼 함수를 추가하여 Place 더미 데이터를 생성
- 각 장소에 주소, 시작 및 종료 시간, 카테고리 등 구체적인 정보를 추가
- `createPlaceImages` 함수를 분리

HJunng committed:
feat: PlaceTimeCard 컴포저블 구현

- 장소와 시간을 표시하는 `PlaceTimeCard` 컴포저블 구현
- 소요 시간에 따라 90분 미만일 경우 `CompactPlaceTimeCard`를, 90분 이상일 경우 `ExpandedPlaceTimeCard`를 표시하도록 구현

HJunng committed:
Merge remote-tracking branch 'refs/remotes/origin/dev' into feature/43-place-component-in-timetable-view

# Conflicts:
#	client/presentation/src/main/java/com/andone/memorip/presentation/util/DummyData.kt

HJunng committed:
rename: CategoryChip을 TagChip으로 이름 변경

- `CategoryChip` -> TagChip으로 이름 변경
- 컴포저블 이름 변경에 따라 `CategoryChipRow`도 `TagChipRow`로 이름 변경

hoyadong1 commented on client/presentation/src/main/java/com/andone/memorip/presentation/theme/Color.kt:
```diff
@@ -67,8 +67,8 @@ internal val darkMemoripColors = MemoripColors(
-    black = Black,
-    white = White,
+    black = White,
+    white = Black,
```
이거 바꾸셨군요 좋습니다

HJunng replied:
이거 다시 바꿔뒀습니다..ㅋㅋㅋ 이렇게 바꿔서 사용하면 다크모드 적용되서 편할 거 같아서 바꿔서 적용해봤는데 꼭 black, white가 필요한 경우도 있을 것 같아서 다시 변경해뒀습니다~

hoyadong1 reviewed (approved):
고생하셨습니다

HJunng committed:
feat: 태그칩 다크모드 대응

- 시스템의 다크/라이트 모드에 따라 태그의 배경색과 대비가 충분히 확보된 텍스트 색상을 사용하도록 `getAdaptiveToneColor` 함수를 추가
- 태그 배경색의 투명도(alpha) 값을 0.1에서 0.15로 조정

uselessnaming commented on client/app/build.gradle.kts:
```diff
@@ -32,6 +32,7 @@ android {
     compileOptions {
         sourceCompatibility = JavaVersion.VERSION_11
         targetCompatibility = JavaVersion.VERSION_11
+        isCoreLibraryDesugaringEnabled = true
```
java time의 min sdk 버전으로 인해서 디슈가링 설정하는 부분인가요?

HJunng replied:
넵 맞습니다!

uselessnaming commented on client/presentation/src/main/java/com/andone/memorip/presentation/plan/component/PlaceTimeCard.kt:
```diff
@@ -0,0 +1,280 @@
+    val EXPANDED_IMAGE_HEIGHT = 160.dp
+    val CARD_ELEVATION = 2.dp
+    val IMAGE_CORNER_RADIUS = 8.dp
+    const val EXPANDED_LAYOUT_THRESHOLD_MINUTES = 90
```
이 값은 Dimen에 있기 조금 애매한 것 같습니다!

HJunng replied:
수정했습니다~

uselessnaming reviewed (approved):
고생하셨습니다!
preview 생성하실 때 기존에 private으로 다 진행해서 이 부분만 신경써주시면 좋을 것 같습니다!

HJunng committed:
fix: 태그칩 UI 개선

- 배경색의 밝기에 따라 텍스트 색상이 검은색 또는 흰색으로 동적으로 결정되도록 수정

HJunng committed:
fix: PlaceTimeCard 한계값 상수로 분리

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/TagChip.kt:
```diff
@@ -0,0 +1,101 @@
+    tag: Category,
+    modifier: Modifier = Modifier,
+) {
+    val textColor = if (tag.color.luminance() > 0.5f) MemoripTheme.colors.black else MemoripTheme.colors.white
```
이 부분도 Dimen으로 분리하면 좋을 것 같습니다!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/TagChip.kt:
```diff
@@ -0,0 +1,101 @@
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.andone.memorip.presentation.selectcategory.model.Category
```
Category를 공통 model로 옮기면 좋을 것 같습니다!

GwonDongHyeon21 reviewed (approved):
고생하셨습니다!
MemoripImage 컴포넌트로 다른 화면의 AsyncImage도 바꾸면 좋을 것 같습니다!
PlaceTimeCard 컴포넌트도 좀 더 분리하면 어떨까요?

HJunng committed:
feat: TagChip 상수 분리

- 색상 밝기(luminance) 임계값을 상수로 추출

HJunng committed:
style: Category 모델의 위치를 공용 model 패키지로 이동

HJunng committed:
feat: 이미지 로딩 로직을 공통 `MemoripImage` 컴포저블로 대체

- `AsyncImage`를 직접 사용하던 부분을 재사용 가능한 `MemoripImage` 컴포저블로 대체

HJunng committed:
feat: 장소 속성 관련 컴포저블 분리

`PlaceTimeCard`에서 사용되던 `TagChipRow`, `PlaceLocationText`, `PlaceTimeText` 컴포저블을 `PlaceAttributeContents.kt` 파일로 분리

HJunng commented:
@GwonDongHyeon21 
피드백 주신 부분 Category 공통 model로 분리, Dimen 분리, AsyncImage 사용되는 거 MemoripImage로 변경 다 해뒀습니다.

PlaceTimeCard 분리는 분리할 만한게 place 속성 관련된 부분이어서 이 부분은 placedetail, placecreate 같은데서도 쓸 수 있을 것 같아서 공용 component로 분리했습니다. 근데 태그들, 주소, 시간 정보들을 분리하는 것보다 Place 관련해서 사용할 때 같이 보는게 좋을 것 같아서 PlaceAttributeContents.kt 파일 하나로 분리해두었습니다~!

<img width="681" height="207" alt="image" src="https://github.com/user-attachments/assets/05d29766-50c7-4566-9279-af6719dbf3f8" />
